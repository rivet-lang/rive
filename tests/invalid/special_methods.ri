struct SpecialMethod {
    func __xxx__() -> {}

    public func __public__(self) {
        _ = self;
    }

    @[inline]
    func __ann__(self) {
        _ = self;
    }
}

func main() -> {}
