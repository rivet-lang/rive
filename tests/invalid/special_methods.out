tests/invalid/special_methods.ri:2:5: error: this special function should be a method
    2 |     func __xxx__() {}
      |     ^~~~~~~~~~~~~~~~~
tests/invalid/special_methods.ri:4:12-6:5: error: special methods cannot be public
      |            /~~ from here
    4 |     public func __public__(self) {
    5 |         _ = self;
    6 |     }
      |     ^~~ to here
tests/invalid/special_methods.ri:9:5-11:5: error: special methods cannot have attributes
      |     /~~ from here
    9 |     func __ann__(self) {
   10 |         _ = self;
   11 |     }
      |     ^~~ to here
rivet: error: could not compile module `special_methods`, aborting due to 3 previous errors